src/directives/toggle/toggle.directive.ts

```diff
@@ -0,0 +1,180 @@
+
+@Directive({
+    exportAs: "toggle",
+    selector: "[igxToggle]"
+})
+export class IgxToggleDirective implements IToggleView, OnInit, OnDestroy {
+
+    @Output()
+    public onOpen = new EventEmitter();
+
+    @Output()
+    public onClose = new EventEmitter();
+
+    @Input()
+    public collapsed = true;
+
+    @Input()
+    public id: string;
+
+    public get element() {
+        return this.elementRef.nativeElement;
+    }
+
+    @HostBinding("class.igx-toggle--hidden")
+    protected get hiddenClass() {
+        return this.collapsed;
+    }
+
+    @HostBinding("class.igx-toggle")
+    protected get defaultClass() {
+        return !this.collapsed;
+    }
+
+    private _id: string;
+
+    constructor(
+        private elementRef: ElementRef,
+        private builder: AnimationBuilder,
+        @Optional() private navigationService: IgxNavigationService) { }
+
+    public open(fireEvents?: boolean, handler?) {
+        if (!this.collapsed) { return; }
+        const player = this.animationActivation();
+        player.onStart(() => this.collapsed = !this.collapsed);
+        player.play();
+        if (fireEvents) {
+            this.onOpen.emit();
+        }
+    }
+
+    public close(fireEvents?: boolean, handler?) {
+        if (this.collapsed) { return; }
+        const player = this.animationActivation();
+        player.onDone(() => this.collapsed = !this.collapsed);
+        player.play();
+        if (fireEvents) {
+            this.onClose.emit();
+        }
+    }
+
+    public toggle(fireEvents?: boolean) {
+        this.collapsed ? this.open(fireEvents) : this.close(fireEvents);
+    }
+
+    public ngOnInit() {
+        if (this.navigationService && this.id) {
+            this.navigationService.add(this.id, this);
+        }
+    }
+
+    public ngOnDestroy() {
+        if (this.navigationService && this.id) {
+            this.navigationService.remove(this.id);
+        }
+    }
+
+    private animationActivation() {
+        let animation: AnimationFactory;
+
+        this.collapsed ?
+            animation = this.openingAnimation() :
+            animation = this.closingAnimation();
+
+        return animation.create(this.elementRef.nativeElement);
+    }
+
+    private openingAnimation() {
+        return this.builder.build([
+            style({ transform: "scaleY(0) translateY(-48px)", transformOrigin: "100% 0%", opacity: 0 }),
+            animate("200ms ease-out", style({ transform: "scaleY(1) translateY(0)", opacity: 1 }))
+        ]);
+    }
+
+    private closingAnimation() {
+        return this.builder.build([
+            style({ transform: "translateY(0)", opacity: 1}),
+            animate("120ms ease-in", style({ transform: "translateY(-12px)", opacity: 0 }))
+        ]);
+    }
+}
+
+@Directive({
+    exportAs: "toggle-action",
+    selector: "[igxToggleAction]"
+})
+export class IgxToggleActionDirective implements OnDestroy, OnInit {
+    @Input()
+    public closeOnOutsideClick = false;
```

Comment:
I think the default expectation is to close the content marked with the directive when clicking outside of it. Did not see anything mentioned about it in the spec but I feel this should be `true` by default. 
@kdinev what is your take on this?